### PR TITLE
For #39721, add progress indicator to project bootstrap

### PR DESCRIPTION
**Pull request by jfboismenu**
Adds a progress indicator to the project bootstrap. This is what it looks like:
image

### CODE REVIEW

File: python/tk_desktop/loading_project_widget.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+
+class LoadingProjectWidget(QtGui.QWidget):
+    """
+    Shows a Shotgun progress widget with a text box containing details of how
+    far along we are in the bootstrap process. It also supports displaying
+    an error message on top.
+    """
+
+    _SHOW_DETAILS = "show details"
```
Comment (manneohrstrom) [inline python/tk_desktop/loading_project_widget.py]: I think we should capitalize these.

File: python/tk_desktop/loading_project_widget.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        self.repaint()
+
+    def paintEvent(self, event):
+        """
+        Draws the overlay.
+        """
+        painter = QtGui.QPainter()
+        painter.begin(self)
+        try:
+            overlay_color = QtGui.QColor("#1B1B1B")
```
Comment (manneohrstrom) [inline python/tk_desktop/loading_project_widget.py]: can this be a core constant?

File: python/tk_desktop/loading_project_widget.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+            painter.setBrush(QtGui.QBrush(overlay_color))
+            painter.setPen(QtGui.QPen(overlay_color))
+            painter.drawRect(0, 0, painter.device().width(), painter.device().height())
+
+            if self._message is not None:
+                # show error text in the center
+                pen = QtGui.QPen(QtGui.QColor("#C8534A"))
```
Comment (manneohrstrom) [inline python/tk_desktop/loading_project_widget.py]: can this be a core constant?

File: python/tk_desktop/loading_project_widget.py (hunk 1 above)
Comment (jfboismenu) [inline python/tk_desktop/loading_project_widget.py]: What do you prefer @robblau ?

screen shot 2017-01-27 at 15 40 57

Comment (robblau) [inline python/tk_desktop/loading_project_widget.py]: "Less details" / "More details"

**Pull request closed without merging**